crates/swc_ecma_transforms_typescript/src/strip.rs

```diff
@@ -1086,32 +1086,186 @@ where
         if module.global || module.declare {
             return None;
         }
-        let module_span = module.span;
 
+        let module_span = module.span;
         let module_name = match module.id {
             TsModuleName::Ident(i) => i,
             TsModuleName::Str(_) => return None,
         };
+
         let body = module.body?;
-        let mut body = match body {
-            TsNamespaceBody::TsModuleBlock(body) => body,
-            TsNamespaceBody::TsNamespaceDecl(_) => return None,
+        let var = self.create_uninit_var(module_name.span, module_name.to_id());
+        let private_name = private_ident!(module_name.sym.clone());
+        let body_stmts = match body {
+            TsNamespaceBody::TsModuleBlock(block) => {
+                self.handle_ts_module_block(block, &private_name)
+            }
+            TsNamespaceBody::TsNamespaceDecl(decl) => {
+                self.handle_ts_namespace_decl(decl, &private_name)
+            }
+        }?;
+
+        self.get_namespace_var_decl_and_call_expr(var, body_stmts, &module_name, &private_name)
+            .map(|(mut var, stmt)| {
+                if let Some(var) = var.as_mut() {
+                    // for comments
+                    var.span = module_span;
+                    // ensure it's a var if top level
+                    if module_name.span.ctxt == self.top_level_ctxt {
+                        var.kind = VarDeclKind::Var;
+                    }
+                }
+                (var.map(Decl::Var), stmt)
+            })
+    }
+
+    fn handle_ts_namespace_decl(
+        &mut self,
+        decl: TsNamespaceDecl,
+        parent_private_name: &Ident,
+    ) -> Option<Vec<Stmt>> {
+        let private_name = private_ident!(decl.id.sym.clone());
+        // we'll always create a variable in this scenario
+        let var_id = Ident::new(
+            decl.id.sym.clone(),
+            DUMMY_SP.with_ctxt(decl.id.span.ctxt.clone()),
```

Comment:
This `.clone()` seems redundant as `SyntaxContext` is `Copy`.

Reply:
Done in 794961694a05c77d529c106ece5ec82a936c096d